Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -300,6 +300,10 @@ private void InjectRequestHeaders(HttpRequestMessage request, string instrumenta
                             {
                                 HttpHeadersUtilities.SetRequestContextKeyValue(requestHeaders, RequestResponseHeaders.RequestContextCorrelationSourceKey, sourceApplicationId);
                             }
+                            else
```

Comment:
Here are the full set of assumptions: 
- If we hit this code, it means we are inside a tracked component.
- We try to recieve appID - it failed.
- No matter why the above happened we want to let the calling component know that they are calling a tracked component.

I don't really care about distinguishing why we failed to get the appID. I think for all cases, tracked dependency should be fine. When instrumentation key is not correct, regardless of whether we log http or http tracked dependency, we have a problem anyways and it can be found with the same degree of ease / difficulty.